sql/src/main/java/io/crate/executor/transport/TransportExecutor.java

```diff
@@ -329,9 +332,25 @@ public Task visitMultiPhasePlan(MultiPhasePlan multiPhasePlan, final Void contex
             Plan rootPlan = multiPhasePlan.rootPlan();
             for (Map.Entry<Plan, SelectSymbol> entry : dependencies.entrySet()) {
                 Plan plan = entry.getKey();
+                SelectSymbol selectSymbol = entry.getValue();
+
+                SelectSymbol.ResultType resultType = selectSymbol.getResultType();
+                SubSelectSymbolReplacer replacer = new SubSelectSymbolReplacer(rootPlan, entry.getValue());
 
-                CollectingBatchConsumer<Collection<Object>, Object[]> consumer =
-                    FirstColumnConsumer.createConsumer();
+                final BatchConsumer consumer;
+                if (resultType == SINGLE_COLUMN_SINGLE_VALUE) {
+                    CollectingBatchConsumer<Object[], Object> consumerImpl =
+                        FirstColumnConsumers.createSingleRowConsumer();
+                    dependencyFutures.add(consumerImpl.resultFuture().thenAccept(replacer::onSuccess));
```

Comment:
Might want to move this logic outside of the if/else


    final CollectingBatchConsumer<?, ?> consumer;
    if (resultType == SINGLE_COLUMN_SINGLE_VALUE) {
        consumer = FirstColumnConsumers.createSingleRowConsumer();
    } else if (resultType == SINGLE_COLUMN_MULTIPLE_VALUES) {
        consumer = FirstColumnConsumers.createConsumer();
    } else {
        throw new IllegalStateException("Can't create consumer: Unknown ResultType");
    }
    SubSelectSymbolReplacer replacer = new SubSelectSymbolReplacer(rootPlan, entry.getValue());
    dependencyFutures.add(consumer.resultFuture().thenAccept(replacer::onSuccess));